### PR TITLE
fix(shorebird_cli): print more friendly error if cli upgrade is required

### LINKED ISSUE #884: fix: message that code push client is out-of-date is not clear
**Description**

When we push a breaking change to the code push protocol, the error message tells the user about `code_push_client`. This is confusing, it should instead tell the user to run `shorebird upgrade`.

### PR DESCRIPTION
**Pull request by bryanoltman**
## Description

When the CodePushClient throws a CodePushUpgradeRequiredException, print an error telling the user to run `shorebird upgrade` instead of the server-provided message about an out of date code push client version.

Screenshot 2023-07-22 at 10 52 24 PM

Fixes https://github.com/shorebirdtech/shorebird/issues/884

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [x] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [x] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: packages/shorebird_cli/lib/src/code_push_client_wrapper.dart [hunk 1]
```diff
@@ -615,4 +618,26 @@ aar artifact already exists, continuing...''',
 
     await promotePatch(appId: appId, patchId: patch.id, channel: channel);
   }
+
+  /// Prints an appropriate error message for the given error and exits with
+  /// code 70. If [progress] is provided, it will be failed with the given
+  /// [message] or [error.toString()] if [message] is null.
+  Never _handleErrorAndExit(
```
Comment (felangel) [inline packages/shorebird_cli/lib/src/code_push_client_wrapper.dart:625]: Couldn’t we have handled the upgrade required case in the code_push_client just like the other exceptions and overridden the message there? I think that would have been a less invasive change and would’ve been more consistent with existing custom error handling?

Comment (bryanoltman) [reply packages/shorebird_cli/lib/src/code_push_client_wrapper.dart:625]: We could have, but my understanding of the code push client design was that it was intentionally decoupled from the CLI, so it wouldn't make sense to reference CLI commands there.

Comment (bryanoltman) [reply packages/shorebird_cli/lib/src/code_push_client_wrapper.dart:625]: (and I think it's important that we tell the user explicitly what command they need to run, as we do elsewhere in the CLI)

Comment (felangel) [reply packages/shorebird_cli/lib/src/code_push_client_wrapper.dart:625]: ah yeah misread this earlier, totally makes sense to have this in the CLI 👍

**Review by felangel — APPROVED** [review]

**Pull request merged**